public/src/js/widgets/config-collection-tags.js

```diff
@@ -0,0 +1,55 @@
+import BaseWidget from 'widgets/base-widget';
+import $ from 'jquery';
+import ko from 'knockout';
+import _ from 'underscore';
+import 'fastsearch';
+import 'fastselect';
+
+export default class CollectionTags extends BaseWidget {
+    constructor(params) {
+        super();
+
+        this.tags = params.tags;
+        const tags = params.tags();
+
+        var initialInput = _.pluck(tags, 'type').join(',');
+
+        this.inputTags = ko.observable(initialInput);
+
+        var initialValues = _.map(tags, tag => {
+            return{
+              'text': tag.type,
+              'value': tag.type
+
+            };
+        });
+
+        $('.multipleInputDynamic').attr('data-initial-value', JSON.stringify(initialValues));
```

Comment:
Just noticed this (separate to this PR): Swap the checkboxes so it's [
checkbox ] <item description> not vice versa.

--
Alastair Jardine, Senior UX Architect

On 15 March 2016 at 15:07, Fabio Crisci <notifications@github.com> wrote:

> In public/src/js/widgets/config-collection-tags.js
> <https://github.com/guardian/facia-tool/pull/142#discussion_r56178586>:
>
> > +        this.tags = params.tags;
> > +        const tags = params.tags();
> > +
> > +        var initialInput = _.pluck(tags, 'type').join(',');
> > +
> > +        this.inputTags = ko.observable(initialInput);
> > +
> > +        var initialValues = _.map(tags, tag => {
> > +            return{
> > +              'text': tag.type,
> > +              'value': tag.type
> > +
> > +            };
> > +        });
> > +
> > +        $('.multipleInputDynamic').attr('data-initial-value', JSON.stringify(initialValues));
>
> this applies to all .multipleInputDynamic in the page, you need to target
> it to a specific element. Look into custom bindings
> <http://knockoutjs.com/documentation/custom-bindings.html>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/guardian/facia-tool/pull/142/files#r56178586
>

-- 

------------------------------
This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396